### PR TITLE
skip vectorization of concatenate op

### PR DESCRIPTION
**Pull request by yunzhongOvO**

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc [hunk 1]
```diff
@@ -555,6 +555,16 @@ struct DiscSpecializeFusionWithSpeculationPass
       return;
     }
 
+    if (isMemIntensiveOptExperimentalEnabled()) {
+      // skip if the root is concatenate operator
+      FusionPatternBase fusion_pattern(fusion_op);
```
Comment (JamesTheZ) [inline tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc:560]: Can be merged with L607.

File: tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc [hunk 2]
```diff
@@ -555,6 +555,16 @@ struct DiscSpecializeFusionWithSpeculationPass
...
+      for (auto root_op : fusion_pattern.getRootOps()) {
+        if (isa<lmhlo::ConcatenateOp>(root_op)) {
+          return ;
```
Comment (JamesTheZ) [inline tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc:563]: Looks like an extra space before `;`. Have you run git-clang-format before the commit?

Comment (yunzhongOvO) [reply tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc]: No. It is done now

File: tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc (hunk 1 above)
Comment (yunzhongOvO) [reply tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc]: Done.

File: tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc [hunk 3]
```diff
@@ -555,6 +555,16 @@ struct DiscSpecializeFusionWithSpeculationPass
       return;
     }
 
+    FusionPatternBase fusion_pattern(fusion_op);
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/transforms/disc_specialize_fusion_with_speculation.cc:558]: It's better to leave some comments about rationales behind.

**Comment by qiuxiafei** [comment]
In addition, the failing tf2_4 CR test seems to be another problem appearing in other PRs. I'm looking into it.

**Review by JamesTheZ — APPROVED** [review]
LGTM.

**Pull request merged**